src/protocol/message.rs

```diff
@@ -236,8 +236,9 @@ impl Message {
     /// Consume the WebSocket and return it as binary data.
     pub fn into_data(self) -> Vec<u8> {
         match self {
-            Message::Text(string) => string.into_bytes(),
-            Message::Binary(data) | Message::Ping(data) | Message::Pong(data) => data,
+            Message::Text(string) => String::from(string).into(),
+            Message::Binary(data) => data.into(),
```

Comment:
I don't think this is an issue in practice, since you're probably only interested in the message data when it's an incoming message, in which case it's always unique (and thus stored as an `Vec<u8>`) anyway.